tests/functionality/pflow/selfcheck.cpp

```diff
@@ -108,23 +132,26 @@ struct PflowFunctionalityTests
   void run_test_case(Params &params) override {
     PetscErrorCode ierr;
     PFLOW pflow;
-    int rank;
-
-    MPI_Comm_rank(PETSC_COMM_WORLD, &rank);
+    char pbuf[PETSC_MAX_PATH_LEN];
+    int my_rank;
+    auto err = MPI_Comm_rank(comm, &my_rank);
+    if (err)
+      throw ExaGOError("Error getting MPI rank number");
 
-    if (rank == 0) {
+    if (my_rank == logging_rank)
       std::cout << "Test Description: " << params.description << std::endl;
-    }
     ierr = PFLOWCreate(params.comm, &pflow);
     ExaGOCheckError(ierr);
 
     // Prepend installation directory to network path
     resolve_datafiles_path(params.network);
+    strncpy(pbuf, params.network.c_str(), params.network.length());
+    pbuf[params.network.length()] = '\0';
```

Comment:
This should either be a part of our API, or removed

Reply:
Okay, let me fool around with this one. I did this to get rid of some memory errors reported by valgrind. I'm not sure whether the c_str() method returns a null-terminated string or just a string that is length() characters long.

Reply:
Maybe related to #31, but we desperately need to use std strings, std filesystem and use C++17.

cc @jaelynlitz

Reply:
I dialed this back and just use std strings and the c_str() function. According to the documentation that I found, c_str() returns null-terminated c strings so I'm not sure why using pbuf gets rid of the memory error. Sometimes valgrind returns stuff that is impossible to figure out. It doesn't look like changing this modifies the behavior of the test suite.

Reply:
Yeah I think valgrind will be hard to sift through without a `~mpi` build, but I appreciate the checks.

Again we have other issues with `.c_str` in the code base, so we might revisit this in another PR anyway

Reply:
Okay, I'm calling this resolved.